mewarpx/mewarpx/setups_store/diode_setup.py

```diff
@@ -589,7 +589,7 @@ def init_simulation(self):
         mwxrun.simulation.max_steps = self.TOTAL_TIMESTEPS
 
         # Add particle species if any were defined
-        if self.SPECIES is not None:
+        if len(self.SPECIES) > 0:
```

Comment:
I believe this will now raise an error if `self.SPECIES` is None. Either make sure self.SPECIES is an empty list if it started out as None, or since python does short-circuit ands, you can do (if not None) and (len > 0) type logic.

Reply:
In the `__init__()` function for the Diode there is a call
```
        # Default null values
        if self.SPECIES is None:
            self.SPECIES = []
```
so we should be good.

Reply:
Ah, good catch, thanks!